core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -262,6 +266,51 @@ public ServerListenerBuilder whenStopped(Iterable<? extends Consumer<? super Ser
         return this;
     }
 
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It will wait indefinitely for the {@link ExecutorService} to terminate during shutdown.
+     */
+    @UnstableApi
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService) {
```

Comment:
Question) This is more a question to @ikhoon since the issue seems to suggest this.
AFAIK `whenStopping` callbacks are invoked before the server starts shutting down.

If the `ExecutorService` is used by the server, I fear that an early cancellation may result in rejected tasks -> failed requests.
Is there a reason we're not adding to `whenStopped` instead?

Reply:
Good question. I think both `shutdownWhenStopping()` and `shutdownWhenStopped()` are useful.
- If an executor is used for periodic jobs such as updating a server's health, an early stop would be preferred for fast-rolling updates.
- If an executor is used for async execution of incoming requests, lazy stopping would make sense.

How about adding both? :-)

Reply:
That sounds like a good idea. I've also added `shutdownWhenStopped()`.